presto-main/src/main/java/com/facebook/presto/connector/informationSchema/InformationSchemaPageSourceProvider.java

```diff
@@ -97,28 +97,16 @@ public ConnectorPageSource createPageSource(ConnectorTransactionHandle transacti
         return new FixedPageSource(pages.build());
     }
 
-    private InternalTable getInternalTable(ConnectorTransactionHandle transactionHandle, ConnectorSession connectorSession, ConnectorSplit connectorSplit, List<ColumnHandle> columns)
+    private InternalTable getInternalTable(ConnectorSession connectorSession, ConnectorSplit connectorSplit, List<ColumnHandle> columns)
     {
-        InformationSchemaTransactionHandle transaction = (InformationSchemaTransactionHandle) transactionHandle;
+        Session session = ((FullConnectorSession) connectorSession).getSession();
```

Comment:
A hack. Not sure how to avoid it though.

Reply:
I looked at it for like 30 minutes and this was the best thing I could come up with also

Reply:
This hack replaces the hack of creating a fake (and incomplete) `Session` from `ConnectorSession` for the special internal connectors. There are some other places that this will allow cleaning up (will do in the future).